File: 0141-LinkedListCycle/LinkedListCycle.java

class LinkedListCycle {
    class Solution {
        public boolean hasCycle(ListNode head) {
            if(head == null || head.next == null) return false;
            
            ListNode slow = head;
            ListNode fast = head.next.next;
            
            while(slow != null && fast != null && fast.next != null){
                if(slow == fast) return true;
                
                slow = slow.next;
                fast = fast.next.next;
            }
            return false;
        }
    }
    
    class ListNode {
        int val;
        ListNode next;
        ListNode(int x) {
            val = x;
            next = null;
        }
    }
}

File: 141-Linked_List_Cycle/Linked_List_Cycle.go
package linkedlistcycle

type ListNode struct {
	Val  int
	Next *ListNode
}

func hasCycle(head *ListNode) bool {
	if head == nil || head.Next == nil {
		return false
	}

	slow := head
	fast := head.Next.Next

	for slow != nil && fast != nil && fast.Next != nil {
		if slow == fast {
			return true
		}

		slow = slow.Next
		fast = fast.Next.Next
	}

	return false
}
